src/field/event_cutscene_celebi.c

```diff
@@ -0,0 +1,245 @@
+#include "demo/legend.naix"
+#include "field/overlay_01_021FB878.h"
+#include "overlay_2/event_cutscene.h"
+
+#include "fieldmap.h"
+#include "follow_mon.h"
+#include "math_util.h"
+#include "overlay_01_021F467C.h"
+#include "overlay_01_021F944C.h"
+#include "overlay_01_022053EC.h"
+#include "script_pokemon_util.h"
+#include "task.h"
+#include "unk_0200FA24.h"
+#include "unk_02062108.h"
+#include "unk_020689C8.h"
+
+typedef struct CelebiTimeTravelCutsceneTaskData {
+    u16 unk0;
+    u8 unk2;
+    u8 isSwirlFinished;
+    u32 unk4;
+    FieldSystem *fieldSystem; // 0x8
+    NNSFndAllocator alloc;    // 0xC
+    Field3dModel model;       // 0x1C
+    Field3DModelAnimation animations[3];
+    Field3dObject object3d;
+    s32 unkE0;
+    u16 unkE4;
+    u16 unkE6;
+    u16 unkE8;
+    u16 perspective;
+    u16 unkEC;
+    u16 unkEE;
+    u8 unkF0;
+    u8 unkF1;
+    u8 frameTimer;
+    u8 unkF3;
+} CelebiTimeTravelCutsceneTaskData;
+
+typedef struct UnkStruct_02253CE0 {
+    u32 unk0;
+    u32 unk4;
+    u16 unk8;
+} UnkStruct_02253CE0;
+
+static BOOL Task_CelebiTimeTravelCutscene(TaskManager *taskMan);
+static void CelebiCutscene_StartSwirlTask(CelebiTimeTravelCutsceneTaskData *data);
+static BOOL ov02_022526EC(CelebiTimeTravelCutsceneTaskData *data);
+static void Task_CelebiCutsceneSwirl(SysTask *task, void *data);
+static void CelebiCutscene_InitSwirlData(CelebiTimeTravelCutsceneTaskData *a0, UnkStruct_02253CE0 *a1);
+static BOOL CelebiCutscene_SwirlEffect(CelebiTimeTravelCutsceneTaskData *data);
+static void Task_CelebiCutsceneSwirlEffect(SysTask *task, void *data);
+static BOOL CelebiCutscene_IsSwirlFinished(CelebiTimeTravelCutsceneTaskData *data);
+static void CelebiCutscene_LoadResources(CelebiTimeTravelCutsceneTaskData *data);
+static void CelebiCutscene_UnloadResources(CelebiTimeTravelCutsceneTaskData *data);
+static void CelebiCutsceneAnimations_FrameSet(Field3DModelAnimation *animations, u32 frame);
+static BOOL CelebiCutsceneAnimations_FrameAdvanceAndCheck(Field3DModelAnimation *animations);
+
+void FieldSystem_BeginCelebiTimeTravelCutsceneTask(FieldSystem *fieldSystem) {
+    CelebiTimeTravelCutsceneTaskData *ptr = AllocFromHeapAtEnd(HEAP_ID_FIELD, sizeof(CelebiTimeTravelCutsceneTaskData));
+    MI_CpuFill8(ptr, 0, sizeof(CelebiTimeTravelCutsceneTaskData));
+    ptr->fieldSystem = fieldSystem;
+    TaskManager_Call(fieldSystem->taskman, Task_CelebiTimeTravelCutscene, ptr);
+}
+
+typedef enum CelebiTimeTravelState {
+    CTT_STATE_BEGIN_PALETTE_FADE,
+    CTT_STATE_LOAD_RESOURCES,
+    CTT_STATE_2,
+    CTT_STATE_SWIRL_EFFECT,
+    CTT_STATE_PALETTE_FADE_2,
+    CTT_STATE_WAIT_UNLOAD_RESOURCES,
+} CelebiTimeTravelState;
+
+static BOOL Task_CelebiTimeTravelCutscene(TaskManager *taskMan) {
+    int *state                             = TaskManager_GetStatePtr(taskMan);
```

Comment:
I believe our .clang-format has changed since you started this file, so it should probably be run again

Reply:
looks like clang-format doesn't catch these anymore, so will fix what I can manually and if I miss some it's nbd